ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarGroupDemoController.swift

```diff
@@ -0,0 +1,289 @@
+        insertAvatarViews(style: .stack, showBorders: false)
+        addTitle(text: "Avatar Stack With Border")
+        insertAvatarViews(style: .stack, showBorders: true)
+        addTitle(text: "Avatar Stack With Mixed Border")
+        insertAvatarViews(style: .stack, showBorders: true, mixed: true)
+        addTitle(text: "Avatar Pile No Border")
+        insertAvatarViews(style: .pile, showBorders: false)
+        addTitle(text: "Avatar Pile With Border")
+        insertAvatarViews(style: .pile, showBorders: true)
+        addTitle(text: "Avatar Pile With Mixed Border")
+        insertAvatarViews(style: .pile, showBorders: true, mixed: true)
+    }
+
+    private struct Constants {
+        static let settingsTextWidth: CGFloat = 200
+        static let avatarsTextWidth: CGFloat = 100
+        static let maxTextInputCharCount: Int = 4
+    }
+
+    private var avatarGroups: [MSFAvatarGroup] = []
+
+    private var avatarCount: Int = 5 {
+        didSet {
+            reloadAvatarViews()
+        }
+    }
+
+    private var isUsingAlternateBackgroundColor: Bool = false {
+        didSet {
+            updateBackgroundColor()
+        }
+    }
+
+    private func updateBackgroundColor() {
+        view.backgroundColor = isUsingAlternateBackgroundColor ? UIColor(light: Colors.gray100, dark: Colors.gray600) : Colors.surfacePrimary
+    }
+
+    @objc private func toggleAlternateBackground(switchView: UISwitch) {
+        isUsingAlternateBackgroundColor = switchView.isOn
+    }
+
+    private var maxDisplayedAvatars: Int = 4 {
+        didSet {
+            if oldValue != maxDisplayedAvatars {
+                maxAvatarsTextField.text = "\(maxDisplayedAvatars)"
+
+                for avatarGroup in avatarGroups {
+                    avatarGroup.state.maxDisplayedAvatars = maxDisplayedAvatars
+                }
+            }
+        }
+    }
+
+    private var overflowCount: Int = 0 {
+        didSet {
+            if oldValue != overflowCount {
+                overflowCountTextField.text = "\(overflowCount)"
+
+                for avatarGroup in avatarGroups {
+                    avatarGroup.state.overflowCount = overflowCount
+                }
+            }
+        }
+    }
+
+    private func insertAvatarViews(style: MSFAvatarGroupStyle, showBorders: Bool, mixed: Bool = false) {
+        var constraints: [NSLayoutConstraint] = []
+
+        for size in MSFAvatarSize.allCases.reversed() {
+            let containerView = UIView(frame: .zero)
+
+            let avatarGroup = MSFAvatarGroup(style: style, size: size)
+            var border = showBorders
+            for index in 0...avatarCount - 1 {
+                vSamplePersonas[index].isRingVisible = border
+                if mixed {
+                    border = !border
+                }
+
+                let avatarState = avatarGroup.state.createAvatar(style: .default, size: size)
+                let samplePersona = vSamplePersonas[index]
```

Comment:
Instead of doing this, the code should get the samplePersonas data class and setting the state properties here:

```
let persona = samplePersonas[index]
state.image = persona.avatarImage
state.primaryText = persona.primaryText
state.secondaryText = persona.secondaryText
```